src/main/java/com/impactupgrade/integration/sfdc/SFDCPartnerAPIClient.java

```diff
@@ -674,8 +675,19 @@ private QueryResult _query(int count, String queryString) throws ConnectionExcep
     try {
       return partnerConnection.get().query(queryString);
     } catch (ApiFault e) {
-      log.error("query failed due to {}: {}", e.getExceptionCode(), e.getExceptionMessage(), e);
-      throw e;
+      if (e instanceof UnexpectedErrorFault && e.getExceptionMessage().contains("Session timed out")) {
+        if (count == 5) {
+          log.error("unable to complete query by attempt {}", count);
+          throw e;
+        }
+        
+        partnerConnection.get().login(authContext.username, authContext.password);
+        return _query(count + 1, queryString);
+
+      } else {
+        log.error("query failed due to {}: {}", e.getExceptionCode(), e.getExceptionMessage(), e);
+        throw e;
+      }
```

Comment:
Also need this fix on insert, update, delete, etc. Maybe there's a way to DRY up the error handling?